solver/utils_test.go

```diff
@@ -145,3 +145,36 @@ bar
 		})
 	}
 }
+
+func Test_IsDocument(t *testing.T) {
+	type args struct {
+		obj interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "document",
+			args: args{
+				obj: &state.Document{},
+			},
+			want: true,
+		},
+		{
+			name: "not document",
+			args: args{
+				obj: &state.Service{},
+			},
```

Comment:
Can we check file.Document or FDocument as well?

Reply:
cb80ca8